// d3-axis-for-react/dist/index.modern.js
import React from 'react';

function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function (target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}

function identity(x) {
  return x;
}
/**
 * Instead of a component for each orientation (like AxisLeft, AxisRight),
 * we provide a value from this Orient object. Provide a value, like
 * Orient.left, to the `orient` prop of the Axis component
 * to place the axis on the left.
 */
var Orient;
(function (Orient) {
  Orient[Orient["top"] = 1] = "top";
  Orient[Orient["right"] = 2] = "right";
  Orient[Orient["bottom"] = 3] = "bottom";
  Orient[Orient["left"] = 4] = "left";
})(Orient || (Orient = {}));
function translateX(x) {
  return "translate(" + x + ",0)";
}
function translateY(y) {
  return "translate(0," + y + ")";
}
/**
 * The axis component. This renders an axis, within a
 * `g` element, for use in a chart.
 */
const Axis = ({
  scale,
  ticks,
  tickArguments: _tickArguments = [],
  tickValues: _tickValues = null,
  tickFormat: _tickFormat = null,
  tickSize,
  tickSizeInner: _tickSizeInner = 6,
  tickSizeOuter: _tickSizeOuter = 6,
  tickPadding: _tickPadding = 3,
  tickTextProps: _tickTextProps = {},
  tickLineProps: _tickLineProps = {},
  domainPathProps: _domainPathProps = {},
  orient: _orient = Orient.bottom,
  offset: _offset = typeof window !== "undefined" && window.devicePixelRatio > 1 ? 0 : 0.5
}) => {
  if (tickSize) {
    _tickSizeInner = tickSize;
    _tickSizeOuter = tickSize;
  }
  if (ticks) {
    _tickArguments = ticks;
  }
  function number(scale) {
    return d => {
      const value = scale(d);
      return value === undefined ? 0 : +value;
    };
  }
  function center(scale, offset) {
    if (scale.bandwidth) {
      offset = Math.max(0, scale.bandwidth() - offset * 2) / 2;
    }
    if (scale.round()) offset = Math.round(offset);
    return d => {
      const value = scale(d);
      return value === undefined ? 0 : value + offset;
    };
  }
  const k = _orient === Orient.top || _orient === Orient.left ? -1 : 1,
    x = _orient === Orient.left || _orient === Orient.right ? "x" : "y",
    transform = _orient === Orient.top || _orient === Orient.bottom ? translateX : translateY;
  // Rendering
  const values = _tickValues == null ? scale.ticks ? scale.ticks.apply(scale, _tickArguments) : scale.domain() : _tickValues,
    format = _tickFormat == null ? "tickFormat" in scale ? scale.tickFormat.apply(scale, _tickArguments) : identity : _tickFormat,
    spacing = Math.max(_tickSizeInner, 0) + _tickPadding,
    range = scale.range(),
    range0 = +range[0] + _offset,
    range1 = +range[range.length - 1] + _offset,
    position = (scale.bandwidth ? center : number)(scale.copy(), _offset);
  const domainPath = _orient === Orient.left || _orient === Orient.right ? _tickSizeOuter ? "M" + k * _tickSizeOuter + "," + range0 + "H" + _offset + "V" + range1 + "H" + k * _tickSizeOuter : "M" + _offset + "," + range0 + "V" + range1 : _tickSizeOuter ? "M" + range0 + "," + k * _tickSizeOuter + "V" + _offset + "H" + range1 + "V" + k * _tickSizeOuter : "M" + range0 + "," + _offset + "H" + range1;
  const lineProps = {
    [x + "2"]: k * _tickSizeInner
  };
  const textProps = {
    [x]: k * spacing
  };
  return /*#__PURE__*/React.createElement("g", null, values.map((tick, i) => /*#__PURE__*/React.createElement("g", {
    className: "tick",
    key: i,
    transform: transform(position(tick) + _offset)
  }, _tickLineProps && /*#__PURE__*/React.createElement("line", _extends({
    stroke: "currentColor"
  }, lineProps, _tickLineProps)), _tickTextProps && /*#__PURE__*/React.createElement("text", _extends({
    fill: "currentColor",
    dy: _orient === Orient.top ? "0em" : _orient === Orient.bottom ? "0.71em" : "0.32em",
    fontSize: "10",
    fontFamily: "sans-serif",
    textAnchor: _orient === Orient.right ? "start" : _orient === Orient.left ? "end" : "middle"
  }, textProps, _tickTextProps), format(tick)))), _domainPathProps && /*#__PURE__*/React.createElement("path", _extends({
    className: "domain",
    stroke: "currentColor",
    fill: "transparent",
    d: domainPath
  }, _domainPathProps)));
};

export { Axis, Orient };
