// d3-axis-for-react/dist/index.es.js
import React from 'react';

function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function (target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}

function identity(x) {
  return x;
}
/**
 * Instead of a component for each orientation (like AxisLeft, AxisRight),
 * we provide a value from this Orient object. Provide a value, like
 * Orient.left, to the `orient` prop of the Axis component
 * to place the axis on the left.
 */
var Orient;
(function (Orient) {
  Orient[Orient["top"] = 1] = "top";
  Orient[Orient["right"] = 2] = "right";
  Orient[Orient["bottom"] = 3] = "bottom";
  Orient[Orient["left"] = 4] = "left";
})(Orient || (Orient = {}));
function translateX(x) {
  return "translate(" + x + ",0)";
}
function translateY(y) {
  return "translate(0," + y + ")";
}
/**
 * The axis component. This renders an axis, within a
 * `g` element, for use in a chart.
 */
var Axis = function Axis(_ref) {
  var _lineProps, _textProps;
  var scale = _ref.scale,
    ticks = _ref.ticks,
    _ref$tickArguments = _ref.tickArguments,
    tickArguments = _ref$tickArguments === void 0 ? [] : _ref$tickArguments,
    _ref$tickValues = _ref.tickValues,
    tickValues = _ref$tickValues === void 0 ? null : _ref$tickValues,
    _ref$tickFormat = _ref.tickFormat,
    tickFormat = _ref$tickFormat === void 0 ? null : _ref$tickFormat,
    tickSize = _ref.tickSize,
    _ref$tickSizeInner = _ref.tickSizeInner,
    tickSizeInner = _ref$tickSizeInner === void 0 ? 6 : _ref$tickSizeInner,
    _ref$tickSizeOuter = _ref.tickSizeOuter,
    tickSizeOuter = _ref$tickSizeOuter === void 0 ? 6 : _ref$tickSizeOuter,
    _ref$tickPadding = _ref.tickPadding,
    tickPadding = _ref$tickPadding === void 0 ? 3 : _ref$tickPadding,
    _ref$tickTextProps = _ref.tickTextProps,
    tickTextProps = _ref$tickTextProps === void 0 ? {} : _ref$tickTextProps,
    _ref$tickLineProps = _ref.tickLineProps,
    tickLineProps = _ref$tickLineProps === void 0 ? {} : _ref$tickLineProps,
    _ref$domainPathProps = _ref.domainPathProps,
    domainPathProps = _ref$domainPathProps === void 0 ? {} : _ref$domainPathProps,
    _ref$orient = _ref.orient,
    orient = _ref$orient === void 0 ? Orient.bottom : _ref$orient,
    _ref$offset = _ref.offset,
    offset = _ref$offset === void 0 ? typeof window !== "undefined" && window.devicePixelRatio > 1 ? 0 : 0.5 : _ref$offset;
  if (tickSize) {
    tickSizeInner = tickSize;
    tickSizeOuter = tickSize;
  }
  if (ticks) {
    tickArguments = ticks;
  }
  function number(scale) {
    return function (d) {
      var value = scale(d);
      return value === undefined ? 0 : +value;
    };
  }
  function center(scale, offset) {
    if (scale.bandwidth) {
      offset = Math.max(0, scale.bandwidth() - offset * 2) / 2;
    }
    if (scale.round()) offset = Math.round(offset);
    return function (d) {
      var value = scale(d);
      return value === undefined ? 0 : value + offset;
    };
  }
  var k = orient === Orient.top || orient === Orient.left ? -1 : 1,
    x = orient === Orient.left || orient === Orient.right ? "x" : "y",
    transform = orient === Orient.top || orient === Orient.bottom ? translateX : translateY;
  // Rendering
  var values = tickValues == null ? scale.ticks ? scale.ticks.apply(scale, tickArguments) : scale.domain() : tickValues,
    format = tickFormat == null ? "tickFormat" in scale ? scale.tickFormat.apply(scale, tickArguments) : identity : tickFormat,
    spacing = Math.max(tickSizeInner, 0) + tickPadding,
    range = scale.range(),
    range0 = +range[0] + offset,
    range1 = +range[range.length - 1] + offset,
    position = (scale.bandwidth ? center : number)(scale.copy(), offset);
  var domainPath = orient === Orient.left || orient === Orient.right ? tickSizeOuter ? "M" + k * tickSizeOuter + "," + range0 + "H" + offset + "V" + range1 + "H" + k * tickSizeOuter : "M" + offset + "," + range0 + "V" + range1 : tickSizeOuter ? "M" + range0 + "," + k * tickSizeOuter + "V" + offset + "H" + range1 + "V" + k * tickSizeOuter : "M" + range0 + "," + offset + "H" + range1;
  var lineProps = (_lineProps = {}, _lineProps[x + "2"] = k * tickSizeInner, _lineProps);
  var textProps = (_textProps = {}, _textProps[x] = k * spacing, _textProps);
  return /*#__PURE__*/React.createElement("g", null, values.map(function (tick, i) {
    return /*#__PURE__*/React.createElement("g", {
      className: "tick",
      key: i,
      transform: transform(position(tick) + offset)
    }, tickLineProps && /*#__PURE__*/React.createElement("line", _extends({
      stroke: "currentColor"
    }, lineProps, tickLineProps)), tickTextProps && /*#__PURE__*/React.createElement("text", _extends({
      fill: "currentColor",
      dy: orient === Orient.top ? "0em" : orient === Orient.bottom ? "0.71em" : "0.32em",
      fontSize: "10",
      fontFamily: "sans-serif",
      textAnchor: orient === Orient.right ? "start" : orient === Orient.left ? "end" : "middle"
    }, textProps, tickTextProps), format(tick)));
  }), domainPathProps && /*#__PURE__*/React.createElement("path", _extends({
    className: "domain",
    stroke: "currentColor",
    fill: "transparent",
    d: domainPath
  }, domainPathProps)));
};

export { Axis, Orient };
