// @dbt-labs/biga/src/processors/formatDocs.ts
import type { TransformedToken } from 'style-dictionary';

import { NON_PRIMITIVE_COLORS } from '../constants.js';
import type { Formatter } from '../types.js';
import { getTokenName } from './utils.js';

/**
 * used for initial output so there is some spacing and its easier to grok pre-commit
 * Prettier will fixup the MD table on commit
 */
const MD_PADDING = ' \t\t ';

type PrimitiveMappings = {
  basic: string[];
  dag: string[];
  viz: string[];
  nav: string[];
};

const defaultPrimitiveMappings: PrimitiveMappings = {
  basic: [],
  dag: [],
  viz: [],
  nav: [],
};

const splitPrimitiveMappings = (mappings: string[]): PrimitiveMappings => {
  return mappings.reduce((acc, curr) => {
    if (/dag/i.test(curr)) {
      acc.dag.push(curr);
    } else if (/viz/i.test(curr)) {
      acc.viz.push(curr);
    } else {
      acc.basic.push(curr);
    }
    return acc;
  }, defaultPrimitiveMappings);
};

const getPrimitiveMappings = (tokens: TransformedToken[]): string[] => {
  const primitiveTokens = tokens.reduce<string[]>((mappings, token) => {
    const unwrapped = token.original.value.replace(/{|}/g, '');
    const cleaned = unwrapped
      .replace(/(color palette|color|primitives|color primitives)\./, '')
      .replace('neutrals.white', 'white')
      .replace('neutrals', 'gray')
      .replace(/(sky blue|sky-blue)/, 'light-blue'); // we used to use light-blue
    const kebabName = cleaned.replace(/\.|\s/g, '-');

    const mode = token.path.includes('light') ? 'light' : 'dark';
    const tokenName = getTokenName(token, mode);
    // avoid also mapping to primitives i.e. we don't want to map `primitivesTeal500` to `active-500`
    if (
      tokenName.includes('primitives') ||
      tokenName.includes('nav') ||
      token.original.value.startsWith('#')
    ) {
      return mappings;
    }
    const modifier = mode === 'dark' ? 'dark:' : '';

    const description = token.description ?? '';
    const mapping = `|${modifier}*-${kebabName} ${MD_PADDING}| ${tokenName} ${MD_PADDING}|${description}${MD_PADDING}|`;
    mappings.push(mapping);
    return mappings;
  }, []);
  return primitiveTokens;
};

/**
 * find any token that has a matching hex value to the non-primitive values to include in mappings
 * builds a table for mappings between `accent` and `active` TW classes and our tokens
 */
const getNonPrimitiveMappings = (tokens: TransformedToken[]): string[] => {
  const nonPrimitiveMappings = tokens.reduce<string[]>((mappings, token) => {
    // exports from figma have a 8bit hex value where our tailwind config only has 6
    const cleanHexValue = token.value.slice(0, 7); // 6 bit hex + leading #
    if (!NON_PRIMITIVE_COLORS[cleanHexValue as keyof typeof NON_PRIMITIVE_COLORS]) {
      return mappings;
    }

    const twClassName =
      NON_PRIMITIVE_COLORS[cleanHexValue as keyof typeof NON_PRIMITIVE_COLORS];
    const mode = token.path.includes('light') ? 'light' : 'dark';
    const tokenName = getTokenName(token, mode);
    // avoid also mapping to primitives i.e. we don't want to map `primitivesTeal500` to `active-500`
    if (tokenName.includes('primitives')) {
      return mappings;
    }
    const modifier = mode === 'dark' ? 'dark:' : '';
    const description = token.description ?? '';
    const mapping = `|${modifier}*-${twClassName} ${MD_PADDING}| ${tokenName}${MD_PADDING}|${description}${MD_PADDING}|`;
    mappings.push(mapping);
    return mappings;
  }, []);

  return nonPrimitiveMappings;
};

const TABLE_HEADER = `
| Tailwind Color${MD_PADDING}| Biga Token${MD_PADDING}| Notes${MD_PADDING}|
|---${MD_PADDING}|---${MD_PADDING}|---${MD_PADDING}|`;

export const formatDocs: Formatter = ({ dictionary, options }) => {
  const primitiveMappings = getPrimitiveMappings(dictionary.allTokens);

  // split the primitive mappings into basic v.s. dag/viz/nav
  const { basic, dag, viz, nav } = splitPrimitiveMappings(primitiveMappings);

  const nonPrimitiveMappings = getNonPrimitiveMappings(dictionary.allTokens);

  return `
# Tailwind to Biga Token Mappings

Simple table to reference as we migrate from TW color classes to tokens.
Most tokens will be found in the first table, but for dag and viz colors, see the second table.
If the TW class does not exist in this table, we probably should not have been using it in the first place OR our design system is missing a token for that color.
In that case, please raise an issue in [#tmp-dev-design-tokens](https://dbt-labs.slack.com/archives/C07TSDLFPM3).

See [TW -> Token Migration](https://www.notion.so/dbtlabs/Tailwind-Token-Migration-12dbb38ebda780e88829d0a3b0ab9fe5?d=12ebb38ebda78012b69f001ce6209e9a#12dbb38ebda780e4b009f828d2048148) for a more information.

${TABLE_HEADER}
${[...basic, ...nonPrimitiveMappings].sort().join('\n')}


## dag & viz Tokens

Our dag and viz tokens require special handling since they utilize more colors than just the basic palette.
See our figma docs for more examples:
- [dag/lineage](https://www.figma.com/design/osn6heqyx0nPtG7s6ti6bE/Pumpernickel-(dbt-cloud-ui)?m=auto&node-id=31372-2218&t=qUgtHCkSzqn9LeFp-1)
- [viz](https://www.figma.com/design/osn6heqyx0nPtG7s6ti6bE/Pumpernickel-(dbt-cloud-ui)?m=auto&node-id=31399-5773&t=qUgtHCkSzqn9LeFp-1)

${TABLE_HEADER}
${[...dag, ...viz, ...nav].sort().join('\n')}

`;
};
