// @dbt-labs/biga/src/processors/utils.ts
import camelCase from 'lodash.camelcase';
import { type TransformedToken } from 'style-dictionary';

import { Mode, type TokensByMode } from '../types.js';

export const splitModes = (tokens: TransformedToken[], modes: Mode[]): TokensByMode => {
  const initialMap: TokensByMode = modes.reduce((acc, mode) => {
    acc[mode] = [];
    return acc;
  }, {} as TokensByMode);

  // split light and dark
  const result = tokens.reduce((acc, curr) => {
    modes.forEach((mode) => {
      if (curr.name.includes(`color-${mode}`) && acc[mode]) {
        acc[mode].push(curr);
      }
      return acc;
    });

    return acc;
  }, initialMap);

  return result;
};

/**
 * will need to extend this later for things like `number` or `typography`
 */
export const getTokenName = (token: TransformedToken, mode: Mode) => {
  const cleaned = token.name.replace(`color-${mode}-`, '');
  return camelCase(cleaned);
};

/**
 * will need to extend this in future to do things like grabbing RGB values from token.attributes
 */
export const getTokenValue = (token: TransformedToken) => {
  return token.value;
};

export const sortByName = (
  tokens: TransformedToken[],
  mode: Mode,
): TransformedToken[] => {
  return tokens.sort((a, b) => {
    return getTokenName(a, mode) < getTokenName(b, mode) ? -1 : 1;
  });
};
