// @dbt-labs/biga/tokens/tokens.js
export const tokens = {
  fgNavMainActive: 'var(--fgNavMainActive)',
  fgNavMainHover: 'var(--fgNavMainHover)',
  fgNavMain: 'var(--fgNavMain)',
  fgNavAccent: 'var(--fgNavAccent)',
  fgNavDecorative: 'var(--fgNavDecorative)',
  fgDuotoneInverse: 'var(--fgDuotoneInverse)',
  fgDuotoneBase: 'var(--fgDuotoneBase)',
  fgMain: 'var(--fgMain)',
  fgWhite: 'var(--fgWhite)',
  fgDisabled: 'var(--fgDisabled)',
  fgBrand: 'var(--fgBrand)',
  fgBrandHover: 'var(--fgBrandHover)',
  fgAlt: 'var(--fgAlt)',
  fgDecorative: 'var(--fgDecorative)',
  fgSuccess: 'var(--fgSuccess)',
  fgSuccessHover: 'var(--fgSuccessHover)',
  fgWarning: 'var(--fgWarning)',
  fgWarningHover: 'var(--fgWarningHover)',
  fgDanger: 'var(--fgDanger)',
  fgDangerHover: 'var(--fgDangerHover)',
  fgBlack: 'var(--fgBlack)',
  fgInfo: 'var(--fgInfo)',
  fgInfoHover: 'var(--fgInfoHover)',
  fgInverse: 'var(--fgInverse)',
  fgBrandActive: 'var(--fgBrandActive)',
  fgSuccessActive: 'var(--fgSuccessActive)',
  fgWarningActive: 'var(--fgWarningActive)',
  fgDangerActive: 'var(--fgDangerActive)',
  fgInfoActive: 'var(--fgInfoActive)',
  fgAltHover: 'var(--fgAltHover)',
  fgAltActive: 'var(--fgAltActive)',
  fgInfoAlt: 'var(--fgInfoAlt)',
  fgInfoAltHover: 'var(--fgInfoAltHover)',
  fgInfoAltActive: 'var(--fgInfoAltActive)',
  fgVizAnalysis: 'var(--fgVizAnalysis)',
  fgVizColumn: 'var(--fgVizColumn)',
  fgVizExposure: 'var(--fgVizExposure)',
  fgVizMacro: 'var(--fgVizMacro)',
  fgVizMetric: 'var(--fgVizMetric)',
  fgVizModel: 'var(--fgVizModel)',
  fgVizSemanticmodel: 'var(--fgVizSemanticmodel)',
  fgVizProject: 'var(--fgVizProject)',
  fgVizSeed: 'var(--fgVizSeed)',
  fgVizSnapshot: 'var(--fgVizSnapshot)',
  fgVizSource: 'var(--fgVizSource)',
  fgVizTest: 'var(--fgVizTest)',
  fgVizSavedquery: 'var(--fgVizSavedquery)',
  fgVizEmerald: 'var(--fgVizEmerald)',
  fgVizRed: 'var(--fgVizRed)',
  fgVizOrange: 'var(--fgVizOrange)',
  fgVizAmber: 'var(--fgVizAmber)',
  fgVizSkyBlue: 'var(--fgVizSkyBlue)',
  fgVizIndigo: 'var(--fgVizIndigo)',
  fgVizTeal: 'var(--fgVizTeal)',
  fgVizNeutral: 'var(--fgVizNeutral)',
  fgVizRedHover: 'var(--fgVizRedHover)',
  fgVizOrangeHover: 'var(--fgVizOrangeHover)',
  fgVizAmberHover: 'var(--fgVizAmberHover)',
  fgVizEmeraldHover: 'var(--fgVizEmeraldHover)',
  fgVizTealHover: 'var(--fgVizTealHover)',
  fgVizSkyBlueHover: 'var(--fgVizSkyBlueHover)',
  fgVizIndigoHover: 'var(--fgVizIndigoHover)',
  fgVizNeutralHover: 'var(--fgVizNeutralHover)',
  fgVizAnalysisMuted: 'var(--fgVizAnalysisMuted)',
  fgVizColumnMuted: 'var(--fgVizColumnMuted)',
  fgVizExposureMuted: 'var(--fgVizExposureMuted)',
  fgVizMacroMuted: 'var(--fgVizMacroMuted)',
  fgVizMetricMuted: 'var(--fgVizMetricMuted)',
  fgVizModelMuted: 'var(--fgVizModelMuted)',
  fgVizSemanticmodelMuted: 'var(--fgVizSemanticmodelMuted)',
  fgVizProjectMuted: 'var(--fgVizProjectMuted)',
  fgVizSeedMuted: 'var(--fgVizSeedMuted)',
  fgVizSnapshotMuted: 'var(--fgVizSnapshotMuted)',
  fgVizSourceMuted: 'var(--fgVizSourceMuted)',
  fgVizTestMuted: 'var(--fgVizTestMuted)',
  fgVizSavedqueryMuted: 'var(--fgVizSavedqueryMuted)',
  fgVizRedMuted: 'var(--fgVizRedMuted)',
  fgVizSkyBlueMuted: 'var(--fgVizSkyBlueMuted)',
  fgVizIndigoMuted: 'var(--fgVizIndigoMuted)',
  fgVizNeutralMuted: 'var(--fgVizNeutralMuted)',
  fgVizOrangeMuted: 'var(--fgVizOrangeMuted)',
  fgVizAmberMuted: 'var(--fgVizAmberMuted)',
  fgVizEmeraldMuted: 'var(--fgVizEmeraldMuted)',
  fgVizTealMuted: 'var(--fgVizTealMuted)',
  fgVizAnalysisHover: 'var(--fgVizAnalysisHover)',
  fgVizColumnHover: 'var(--fgVizColumnHover)',
  fgVizExposureHover: 'var(--fgVizExposureHover)',
  fgVizMacroHover: 'var(--fgVizMacroHover)',
  fgVizMetricHover: 'var(--fgVizMetricHover)',
  fgVizModelHover: 'var(--fgVizModelHover)',
  fgVizSemanticmodelHover: 'var(--fgVizSemanticmodelHover)',
  fgVizProjectHover: 'var(--fgVizProjectHover)',
  fgVizSeedHover: 'var(--fgVizSeedHover)',
  fgVizSnapshotHover: 'var(--fgVizSnapshotHover)',
  fgVizSourceHover: 'var(--fgVizSourceHover)',
  fgVizTestHover: 'var(--fgVizTestHover)',
  fgVizSavedqueryHover: 'var(--fgVizSavedqueryHover)',
  fgVizMonoBrand1: 'var(--fgVizMonoBrand1)',
  fgVizMonoBrand2: 'var(--fgVizMonoBrand2)',
  fgVizMonoBrand3: 'var(--fgVizMonoBrand3)',
  fgVizMonoBrand4: 'var(--fgVizMonoBrand4)',
  fgVizMonoBrand5: 'var(--fgVizMonoBrand5)',
  fgVizMonoBrand6: 'var(--fgVizMonoBrand6)',
  fgVizMonoBrand7: 'var(--fgVizMonoBrand7)',
  fgVizMonoBrand8: 'var(--fgVizMonoBrand8)',
  fgVizMonoBrand9: 'var(--fgVizMonoBrand9)',
  fgVizFunction: 'var(--fgVizFunction)',
  fgVizFunctionHover: 'var(--fgVizFunctionHover)',
  fgVizFunctionMuted: 'var(--fgVizFunctionMuted)',
  fgBrandAlt: 'var(--fgBrandAlt)',
  fgBrandAltHover: 'var(--fgBrandAltHover)',
  fgBrandAltActive: 'var(--fgBrandAltActive)',
  fgSyntaxVariables: 'var(--fgSyntaxVariables)',
  fgSyntaxKeywords: 'var(--fgSyntaxKeywords)',
  fgSyntaxNumbers: 'var(--fgSyntaxNumbers)',
  fgSyntaxClasses: 'var(--fgSyntaxClasses)',
  fgSyntaxFunctions: 'var(--fgSyntaxFunctions)',
  fgSyntaxStrings: 'var(--fgSyntaxStrings)',
  fgSyntaxOperators: 'var(--fgSyntaxOperators)',
  fgSyntaxComments: 'var(--fgSyntaxComments)',
  fgSyntaxPunctuation: 'var(--fgSyntaxPunctuation)',
  fgSyntaxConstants: 'var(--fgSyntaxConstants)',
  fgBrandSelectionHover: 'var(--fgBrandSelectionHover)',
  fgBrandSelectionActive: 'var(--fgBrandSelectionActive)',
  fgBrandSelection: 'var(--fgBrandSelection)',
  fgNodeOutput: 'var(--fgNodeOutput)',
  fgNodeOutputHover: 'var(--fgNodeOutputHover)',
  fgNodeTransform: 'var(--fgNodeTransform)',
  fgNodeTransformHover: 'var(--fgNodeTransformHover)',
  fgNodeInput: 'var(--fgNodeInput)',
  fgNodeInputHover: 'var(--fgNodeInputHover)',
  fgNodeModify: 'var(--fgNodeModify)',
  fgNodeModifyHover: 'var(--fgNodeModifyHover)',
  fgAnsiBlack: 'var(--fgAnsiBlack)',
  fgAnsiBlackBright: 'var(--fgAnsiBlackBright)',
  fgAnsiBlue: 'var(--fgAnsiBlue)',
  fgAnsiCyan: 'var(--fgAnsiCyan)',
  fgAnsiCyanBright: 'var(--fgAnsiCyanBright)',
  fgAnsiGreen: 'var(--fgAnsiGreen)',
  fgAnsiGreenBright: 'var(--fgAnsiGreenBright)',
  fgAnsiMagenta: 'var(--fgAnsiMagenta)',
  fgAnsiMagentaBright: 'var(--fgAnsiMagentaBright)',
  fgAnsiYellow: 'var(--fgAnsiYellow)',
  fgAnsiYellowBright: 'var(--fgAnsiYellowBright)',
  fgAnsiRed: 'var(--fgAnsiRed)',
  fgAnsiRedBright: 'var(--fgAnsiRedBright)',
  fgAnsiWhite: 'var(--fgAnsiWhite)',
  fgAnsiWhiteBright: 'var(--fgAnsiWhiteBright)',
  fgAnsiMain: 'var(--fgAnsiMain)',
  fgAnsiBlueBright: 'var(--fgAnsiBlueBright)',
  fgAnsiDimmed: 'var(--fgAnsiDimmed)',
  bgNavTag: 'var(--bgNavTag)',
  bgNavHeader: 'var(--bgNavHeader)',
  bgNavMainActive: 'var(--bgNavMainActive)',
  bgNavMainHover: 'var(--bgNavMainHover)',
  bgNavMain: 'var(--bgNavMain)',
  bgNavRaised: 'var(--bgNavRaised)',
  bgDagAnalysis: 'var(--bgDagAnalysis)',
  bgDagColumn: 'var(--bgDagColumn)',
  bgDagExposure: 'var(--bgDagExposure)',
  bgDagMacro: 'var(--bgDagMacro)',
  bgDagMetric: 'var(--bgDagMetric)',
  bgDagModel: 'var(--bgDagModel)',
  bgDagSemanticmodel: 'var(--bgDagSemanticmodel)',
  bgDagProject: 'var(--bgDagProject)',
  bgDagSeed: 'var(--bgDagSeed)',
  bgDagSnapshot: 'var(--bgDagSnapshot)',
  bgDagSource: 'var(--bgDagSource)',
  bgDagTest: 'var(--bgDagTest)',
  bgDagSavedquery: 'var(--bgDagSavedquery)',
  bgDagFunction: 'var(--bgDagFunction)',
  bgMain: 'var(--bgMain)',
  bgBlack: 'var(--bgBlack)',
  bgMainHover: 'var(--bgMainHover)',
  bgDisabled: 'var(--bgDisabled)',
  bgBrand: 'var(--bgBrand)',
  bgBrandHover: 'var(--bgBrandHover)',
  bgDanger: 'var(--bgDanger)',
  bgDangerHover: 'var(--bgDangerHover)',
  bgSuccessMuted: 'var(--bgSuccessMuted)',
  bgWarningMuted: 'var(--bgWarningMuted)',
  bgDangerMuted: 'var(--bgDangerMuted)',
  bgNeutralMuted: 'var(--bgNeutralMuted)',
  bgBackground: 'var(--bgBackground)',
  bgInfoMuted: 'var(--bgInfoMuted)',
  bgWhite: 'var(--bgWhite)',
  bgDuotoneSuccess: 'var(--bgDuotoneSuccess)',
  bgDuotoneNeutral: 'var(--bgDuotoneNeutral)',
  bgDuotoneMuted: 'var(--bgDuotoneMuted)',
  bgDuotoneDanger: 'var(--bgDuotoneDanger)',
  bgDuotoneWarning: 'var(--bgDuotoneWarning)',
  bgDuotoneInfo: 'var(--bgDuotoneInfo)',
  bgDuotoneInfoAlt: 'var(--bgDuotoneInfoAlt)',
  bgSuccessHover: 'var(--bgSuccessHover)',
  bgWarningHover: 'var(--bgWarningHover)',
  bgInfoHover: 'var(--bgInfoHover)',
  bgNeutralHover: 'var(--bgNeutralHover)',
  bgSuccess: 'var(--bgSuccess)',
  bgWarning: 'var(--bgWarning)',
  bgBrandMuted: 'var(--bgBrandMuted)',
  bgInfo: 'var(--bgInfo)',
  bgBadgeRedMuted: 'var(--bgBadgeRedMuted)',
  bgBadgeOrangeMuted: 'var(--bgBadgeOrangeMuted)',
  bgBadgeAmberMuted: 'var(--bgBadgeAmberMuted)',
  bgBadgeEmeraldMuted: 'var(--bgBadgeEmeraldMuted)',
  bgBadgeSkyBlueMuted: 'var(--bgBadgeSkyBlueMuted)',
  bgBadgeIndigoMuted: 'var(--bgBadgeIndigoMuted)',
  bgBadgeNeutralMuted: 'var(--bgBadgeNeutralMuted)',
  bgBadgeBrandMuted: 'var(--bgBadgeBrandMuted)',
  bgBadgeTealMuted: 'var(--bgBadgeTealMuted)',
  bgInverse: 'var(--bgInverse)',
  bgInverseHover: 'var(--bgInverseHover)',
  bgNeutral: 'var(--bgNeutral)',
  bgMainActive: 'var(--bgMainActive)',
  bgInverseActive: 'var(--bgInverseActive)',
  bgBrandActive: 'var(--bgBrandActive)',
  bgSuccessActive: 'var(--bgSuccessActive)',
  bgWarningActive: 'var(--bgWarningActive)',
  bgDangerActive: 'var(--bgDangerActive)',
  bgInfoActive: 'var(--bgInfoActive)',
  bgNeutralActive: 'var(--bgNeutralActive)',
  bgBrandTinted: 'var(--bgBrandTinted)',
  bgSuccessTinted: 'var(--bgSuccessTinted)',
  bgWarningTinted: 'var(--bgWarningTinted)',
  bgDangerTinted: 'var(--bgDangerTinted)',
  bgInfoTinted: 'var(--bgInfoTinted)',
  bgNeutralTinted: 'var(--bgNeutralTinted)',
  bgInfoAlt: 'var(--bgInfoAlt)',
  bgInfoAltHover: 'var(--bgInfoAltHover)',
  bgInfoAltActive: 'var(--bgInfoAltActive)',
  bgInfoAltMuted: 'var(--bgInfoAltMuted)',
  bgBrandAlt: 'var(--bgBrandAlt)',
  bgBrandAltHover: 'var(--bgBrandAltHover)',
  bgBrandAltActive: 'var(--bgBrandAltActive)',
  bgBrandAltMuted: 'var(--bgBrandAltMuted)',
  bgOverlayTinted: 'var(--bgOverlayTinted)',
  bgMainHoverTinted: 'var(--bgMainHoverTinted)',
  bgInfoAltTinted: 'var(--bgInfoAltTinted)',
  bgBrandSelection: 'var(--bgBrandSelection)',
  bgBrandSelectionHover: 'var(--bgBrandSelectionHover)',
  bgBrandSelectionActive: 'var(--bgBrandSelectionActive)',
  bgNodeOutputMuted: 'var(--bgNodeOutputMuted)',
  bgNodeTransformMuted: 'var(--bgNodeTransformMuted)',
  bgNodeInputMuted: 'var(--bgNodeInputMuted)',
  bgNodeModifyMuted: 'var(--bgNodeModifyMuted)',
  bgCanvasBackground: 'var(--bgCanvasBackground)',
  bgCanvasDot: 'var(--bgCanvasDot)',
  bgRaised: 'var(--bgRaised)',
  bgBrandAltTinted: 'var(--bgBrandAltTinted)',
  borderResizableHover: 'var(--borderResizableHover)',
  borderResizableActive: 'var(--borderResizableActive)',
  borderMuted: 'var(--borderMuted)',
  borderMain: 'var(--borderMain)',
  borderBrandHover: 'var(--borderBrandHover)',
  borderDanger: 'var(--borderDanger)',
  borderSuccessMuted: 'var(--borderSuccessMuted)',
  borderWarningMuted: 'var(--borderWarningMuted)',
  borderDangerMuted: 'var(--borderDangerMuted)',
  borderBrandAltMuted: 'var(--borderBrandAltMuted)',
  borderNeutralMuted: 'var(--borderNeutralMuted)',
  borderDisabled: 'var(--borderDisabled)',
  borderMainHover: 'var(--borderMainHover)',
  borderInfoMuted: 'var(--borderInfoMuted)',
  borderBrand: 'var(--borderBrand)',
  borderSuccess: 'var(--borderSuccess)',
  borderWarning: 'var(--borderWarning)',
  borderInfo: 'var(--borderInfo)',
  borderDangerHover: 'var(--borderDangerHover)',
  borderBrandMuted: 'var(--borderBrandMuted)',
  borderSuccessHover: 'var(--borderSuccessHover)',
  borderWarningHover: 'var(--borderWarningHover)',
  borderInfoHover: 'var(--borderInfoHover)',
  borderBadgeRedMuted: 'var(--borderBadgeRedMuted)',
  borderBadgeOrangeMuted: 'var(--borderBadgeOrangeMuted)',
  borderBadgeAmberMuted: 'var(--borderBadgeAmberMuted)',
  borderBadgeEmeraldMuted: 'var(--borderBadgeEmeraldMuted)',
  borderBadgeSkyBlueMuted: 'var(--borderBadgeSkyBlueMuted)',
  borderBadgeIndigoMuted: 'var(--borderBadgeIndigoMuted)',
  borderBadgeNeutralMuted: 'var(--borderBadgeNeutralMuted)',
  borderBadgeBrandMuted: 'var(--borderBadgeBrandMuted)',
  borderMainActive: 'var(--borderMainActive)',
  borderBrandActive: 'var(--borderBrandActive)',
  borderDangerActive: 'var(--borderDangerActive)',
  borderSuccessActive: 'var(--borderSuccessActive)',
  borderWarningActive: 'var(--borderWarningActive)',
  borderInfoActive: 'var(--borderInfoActive)',
  borderNavMuted: 'var(--borderNavMuted)',
  borderInfoAlt: 'var(--borderInfoAlt)',
  borderInfoAltHover: 'var(--borderInfoAltHover)',
  borderInfoAltActive: 'var(--borderInfoAltActive)',
  borderInfoAltMuted: 'var(--borderInfoAltMuted)',
  borderBrandAlt: 'var(--borderBrandAlt)',
  borderBrandAltHover: 'var(--borderBrandAltHover)',
  borderBrandAltActive: 'var(--borderBrandAltActive)',
  borderVizRed: 'var(--borderVizRed)',
  borderVizOrange: 'var(--borderVizOrange)',
  borderVizAmber: 'var(--borderVizAmber)',
  borderVizEmerald: 'var(--borderVizEmerald)',
  borderVizTeal: 'var(--borderVizTeal)',
  borderVizSkyBlue: 'var(--borderVizSkyBlue)',
  borderVizIndigo: 'var(--borderVizIndigo)',
  borderVizNeutral: 'var(--borderVizNeutral)',
  borderVizRedHover: 'var(--borderVizRedHover)',
  borderVizOrangeHover: 'var(--borderVizOrangeHover)',
  borderVizAmberHover: 'var(--borderVizAmberHover)',
  borderVizEmeraldHover: 'var(--borderVizEmeraldHover)',
  borderVizTealHover: 'var(--borderVizTealHover)',
  borderVizSkyBlueHover: 'var(--borderVizSkyBlueHover)',
  borderVizIndigoHover: 'var(--borderVizIndigoHover)',
  borderVizNeutralHover: 'var(--borderVizNeutralHover)',
  borderVizAnalysis: 'var(--borderVizAnalysis)',
  borderVizAnalysisHover: 'var(--borderVizAnalysisHover)',
  borderVizColumn: 'var(--borderVizColumn)',
  borderVizColumnHover: 'var(--borderVizColumnHover)',
  borderVizExposure: 'var(--borderVizExposure)',
  borderVizExposureHover: 'var(--borderVizExposureHover)',
  borderVizMacro: 'var(--borderVizMacro)',
  borderVizMacroHover: 'var(--borderVizMacroHover)',
  borderVizMetric: 'var(--borderVizMetric)',
  borderVizMetricHover: 'var(--borderVizMetricHover)',
  borderVizModel: 'var(--borderVizModel)',
  borderVizModelHover: 'var(--borderVizModelHover)',
  borderVizSemanticmodel: 'var(--borderVizSemanticmodel)',
  borderVizSemanticmodelHover: 'var(--borderVizSemanticmodelHover)',
  borderVizProject: 'var(--borderVizProject)',
  borderVizProjectHover: 'var(--borderVizProjectHover)',
  borderVizSeed: 'var(--borderVizSeed)',
  borderVizSeedHover: 'var(--borderVizSeedHover)',
  borderVizSnapshot: 'var(--borderVizSnapshot)',
  borderVizSnapshotHover: 'var(--borderVizSnapshotHover)',
  borderVizSource: 'var(--borderVizSource)',
  borderVizSourceHover: 'var(--borderVizSourceHover)',
  borderVizTest: 'var(--borderVizTest)',
  borderVizTestHover: 'var(--borderVizTestHover)',
  borderVizSavedquery: 'var(--borderVizSavedquery)',
  borderVizSavedqueryHover: 'var(--borderVizSavedqueryHover)',
  borderVizRedMuted: 'var(--borderVizRedMuted)',
  borderVizOrangeMuted: 'var(--borderVizOrangeMuted)',
  borderVizTealMuted: 'var(--borderVizTealMuted)',
  borderVizSkyBlueMuted: 'var(--borderVizSkyBlueMuted)',
  borderVizIndigoMuted: 'var(--borderVizIndigoMuted)',
  borderVizNeutralMuted: 'var(--borderVizNeutralMuted)',
  borderVizAnalysisMuted: 'var(--borderVizAnalysisMuted)',
  borderVizColumnMuted: 'var(--borderVizColumnMuted)',
  borderVizExposureMuted: 'var(--borderVizExposureMuted)',
  borderVizMacroMuted: 'var(--borderVizMacroMuted)',
  borderVizMetricMuted: 'var(--borderVizMetricMuted)',
  borderVizModelMuted: 'var(--borderVizModelMuted)',
  borderVizSemanticmodelMuted: 'var(--borderVizSemanticmodelMuted)',
  borderVizProjectMuted: 'var(--borderVizProjectMuted)',
  borderVizSeedMuted: 'var(--borderVizSeedMuted)',
  borderVizSnapshotMuted: 'var(--borderVizSnapshotMuted)',
  borderVizSourceMuted: 'var(--borderVizSourceMuted)',
  borderVizTestMuted: 'var(--borderVizTestMuted)',
  borderVizAmberMuted: 'var(--borderVizAmberMuted)',
  borderVizEmeraldMuted: 'var(--borderVizEmeraldMuted)',
  borderVizSavedqueryMuted: 'var(--borderVizSavedqueryMuted)',
  borderVizMonoBrand1: 'var(--borderVizMonoBrand1)',
  borderVizMonoBrand2: 'var(--borderVizMonoBrand2)',
  borderVizMonoBrand3: 'var(--borderVizMonoBrand3)',
  borderVizMonoBrand4: 'var(--borderVizMonoBrand4)',
  borderVizMonoBrand5: 'var(--borderVizMonoBrand5)',
  borderVizMonoBrand6: 'var(--borderVizMonoBrand6)',
  borderVizMonoBrand7: 'var(--borderVizMonoBrand7)',
  borderVizMonoBrand8: 'var(--borderVizMonoBrand8)',
  borderVizMonoBrand9: 'var(--borderVizMonoBrand9)',
  borderVizFunction: 'var(--borderVizFunction)',
  borderVizFunctionHover: 'var(--borderVizFunctionHover)',
  borderVizFunctionMuted: 'var(--borderVizFunctionMuted)',
  borderTinted: 'var(--borderTinted)',
  focusHalo: 'var(--focusHalo)',
  focusDangerhalo: 'var(--focusDangerhalo)',
  focusBrand: 'var(--focusBrand)',
  focusDanger: 'var(--focusDanger)',
  focusInverse: 'var(--focusInverse)',
  focusNeutral: 'var(--focusNeutral)',
  focusDangerTinted: 'var(--focusDangerTinted)',
  elevationDanger: 'var(--elevationDanger)',
  elevationBrand: 'var(--elevationBrand)',
  elevationWarning: 'var(--elevationWarning)',
  elevationNodeEmphasis: 'var(--elevationNodeEmphasis)',
  elevationNodeMuted: 'var(--elevationNodeMuted)',
  elevationDangerMuted: 'var(--elevationDangerMuted)',
  elevationDangerBase: 'var(--elevationDangerBase)',
  elevationDangerEmphasis: 'var(--elevationDangerEmphasis)',
  elevationBase: 'var(--elevationBase)',
  elevationInner: 'var(--elevationInner)',
  elevationNode: 'var(--elevationNode)',
  elevationSuccess: 'var(--elevationSuccess)',
  elevationInfo: 'var(--elevationInfo)',
  logoBit: 'var(--logoBit)',
  logoLogotype: 'var(--logoLogotype)',
  brandOrange: 'var(--brandOrange)',
  brandBlack: 'var(--brandBlack)',
  brandPurple: 'var(--brandPurple)',
  brandPink: 'var(--brandPink)',
  brandGreen: 'var(--brandGreen)',
  brandBlue: 'var(--brandBlue)',
  brandYellow: 'var(--brandYellow)',
  brandOrangeTinted: 'var(--brandOrangeTinted)',
  brandPurpleTinted: 'var(--brandPurpleTinted)',
  brandPinkTinted: 'var(--brandPinkTinted)',
  brandGreenTinted: 'var(--brandGreenTinted)',
  brandBlueTinted: 'var(--brandBlueTinted)',
  brandYellowTinted: 'var(--brandYellowTinted)',
  brandYellowTransparent: 'var(--brandYellowTransparent)',
  brandGreenTransparent: 'var(--brandGreenTransparent)',
  brandBlueTransparent: 'var(--brandBlueTransparent)',
  brandOrangeTransparent: 'var(--brandOrangeTransparent)',
  brandPurpleTransparent: 'var(--brandPurpleTransparent)',
  brandPinkTransparent: 'var(--brandPinkTransparent)',
};
