// @dbt-labs/sourdough/eslint.config.mjs
import { defineConfig } from 'eslint/config';
import baseConfig from '@dbt-labs/eslint-config/react';
import bigaPlugin from '@dbt-labs/eslint-plugin-biga';
import sentenceCasePlugin from 'eslint-plugin-sentence-case';

export default defineConfig([
  ...baseConfig,
  {
    plugins: {
      '@dbt-labs/biga': bigaPlugin,
      'sentence-case': sentenceCasePlugin,
    },
  },
  {
    files: ['src/**/*.[jt]s?(x)'],
    rules: {
      '@typescript-eslint/no-unused-vars': [
        'error',
        {
          ignoreRestSiblings: true,
          argsIgnorePattern: '^_',
          destructuredArrayIgnorePattern: '^_',
        },
      ],
    },
    ignores: ['src/**/*.stories.*'],
  },
  {
    files: ['src/**/*.stories.*', 'scripts/**/*', 'cypress/**/*'],
    rules: {
      'no-console': 'off',
    },
  },
  {
    files: ['**/*.spec.ts', '**/*.spec.tsx', '**/*.test.ts', '**/*.test.tsx'],
    rules: {
      'no-restricted-properties': [
        'error',
        {
          object: 'describe',
          property: 'only',
          message: 'Do not commit focused tests with describe.only',
        },
        {
          object: 'it',
          property: 'only',
          message: 'Do not commit focused tests with it.only',
        },
        {
          object: 'test',
          property: 'only',
          message: 'Do not commit focused tests with test.only',
        },
        {
          object: 'context',
          property: 'only',
          message: 'Do not commit focused tests with context.only',
        },
      ],
    },
  },
  {
    files: ['src/**/*.[jt]s?(x)'],
    rules: {
      '@dbt-labs/biga/no-invalid-biga-tokens': 'warn',
      // '@dbt-labs/biga/no-tw-color-classes': 'warn',
      'sentence-case/sentence-case': 'warn',
    },
  },
]);
