// @dbt-labs/sourdough/.storybook/main.ts
import { dirname, join } from 'path';
import type { StorybookConfig } from '@storybook/react-vite';
import remarkGfm from 'remark-gfm';

// Detect if running in CI environment
const isCI = process.env.CI === 'true';

const config: StorybookConfig = {
  stories: ['../src/**/*.docs.mdx', '../src/**/*.stories.@(ts|tsx)'],
  addons: [
    {
      name: '@storybook/addon-essentials',
      options: {
        backgrounds: false,
      },
    },
    {
      name: '@storybook/addon-docs',
      options: {
        mdxPluginOptions: {
          mdxCompileOptions: {
            remarkPlugins: [remarkGfm],
          },
        },
      },
    },
    getAbsolutePath('@storybook/addon-themes'),
    getAbsolutePath('@storybook/addon-storysource'),
  ],
  framework: {
    name: getAbsolutePath('@storybook/react-vite'),
    options: {},
  },
  core: {
    builder: getAbsolutePath('@storybook/builder-vite'),
    disableTelemetry: true,
    // Disable manager UI in CI environments
    ...(isCI && {
      disableTelemetry: true,
      enableCrashReports: false,
    }),
  },
  typescript: {
    reactDocgen: 'react-docgen-typescript',
  },
  managerHead: (head) => `
          ${head}
          <meta name="robots" content="noindex" />
          ${
            isCI
              ? '<style>.sb-bar, .css-1d1nb5w { display: none !important; }</style>'
              : ''
          }
        `,
  viteFinal: (_config) => ({
    ..._config,
    optimizeDeps: {
      ..._config.optimizeDeps,
      exclude: ['*sb-vite', '*storybook', '*react-hot-toast'],
    },
  }),
};
export default config;

function getAbsolutePath(value: string): any {
  return dirname(require.resolve(join(value, 'package.json')));
}
