// @dbt-labs/sourdough/.storybook/preview.tsx
import { withThemeByClassName } from '@storybook/addon-themes';
import type { Decorator, Preview, ReactRenderer } from '@storybook/react';

import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { ThemeProvider, useTheme } from '../src/context';
import { AutoDocsTemplate } from '../src/docs/AutoDocsTemplate';
import '../src/sourdough.css';
import '@dbt-labs/biga/tokens.css'; // adds token css variables

// adds theme toggle to storybook toolbar
const sbThemeDecorator: Decorator = withThemeByClassName<ReactRenderer>({
  themes: {
    light: 'light',
    dark: 'dark',
  },
  defaultTheme: 'light',
});

const queryClient = new QueryClient();

/**
 * listen for theme changes in the url and update the theme context
 */
const ThemeUpdater = () => {
  const { setTheme } = useTheme();
  const searchParams = new URLSearchParams(window.location.search);
  const isDark = searchParams.get('globals')?.includes('theme:dark');
  const nextTheme = isDark ? 'dark' : 'light';
  setTheme(nextTheme);
  return null;
};

const previewWrapper: Decorator = (Story, context) => {
  // overriding SB's default styles and allowing .dark class to extend to edges
  const storyStyle = {
    margin: -20,
    padding: 20,
    minHeight: '100vh',
    height: '100%',
    backgroundColor: 'var(--bgMain)',
    color: 'var(--fgMain)',
  };
  const style = context.viewMode === 'docs' ? {} : storyStyle;
  return (
    <ThemeProvider isEnabled={true} shouldStoreLocally={true}>
      <ThemeUpdater />
      <div className="rounded-lg p-6" style={style}>
        <div className="p-6">
          <QueryClientProvider client={queryClient}>
            <MemoryRouter initialEntries={['/']}>
              <Story />
            </MemoryRouter>
          </QueryClientProvider>
        </div>
      </div>
    </ThemeProvider>
  );
};

const decorators: Decorator[] = [
  // needs to go first in order for hooks to work!
  // this is due to hooks needing to be in a <Story /> function comp
  previewWrapper,
  sbThemeDecorator,
];

// pulled from tailwind theme.screens
const viewports = {
  small: {
    name: 'small',
    styles: {
      width: '800px',
      height: '100%',
    },
    type: 'mobile',
  },
  medium: {
    name: 'medium',
    styles: {
      width: '960px',
      height: '100%',
    },
    type: 'tablet',
  },
  large: {
    name: 'large',
    styles: {
      width: '100%',
      height: '100%',
    },
    type: 'desktop',
  },
};

const preview: Preview = {
  parameters: {
    controls: {
      sort: 'requiredFirst',
      matchers: {
        color: /(background|color)$/i,
        date: /Date$/i,
      },
      expanded: false,
    },
    docs: {
      page: AutoDocsTemplate,
      controls: {
        sort: 'requiredFirst',
      },
    },
    // toolbar items from addon-essentials
    viewport: { viewports },

    options: {
      storySort: {
        method: 'alphabetical',
        order: [
          'Docs',
          [
            'Getting Started',
            'Contributing',
            'Documenting Components',
            'Cloud UI Code Base',
            'Libraries',
            '*',
          ],
          'Styles',
          'Elements',
          ['Intro'],
          'Forms',
          ['Building Blocks', 'Fields'],
          'Layouts',
          ['Intro', 'Building Blocks'],
          'Examples',
          ['Loading Data', 'Saving Data', 'Building Forms'],
          'Utilities',
          ['Intro'],
          'Data Viz',
          'Deprecated',
        ],
      },
    },
  },
  decorators,
};

export default preview;
